pyccel/codegen/printing/ccode.py

```diff
@@ -1466,16 +1466,92 @@ def _print_EmptyNode(self, expr):
     #=================== OMP ==================
     def _print_OMP_For_Loop(self, expr):
         omp_expr   = str(expr.txt)
-        return '#pragma omp for{}\n{{'.format(omp_expr)
+        return '#pragma omp for{}'.format(omp_expr)
 
     def _print_OMP_Parallel_Construct(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp {}\n{{'.format(omp_expr)
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr   = '#pragma omp parallel{}'.format(clauses)
+        if expr.combined is None:
+            omp_expr += '\n{'
+        elif (expr.combined and "for" not in expr.combined):
+            if "masked taskloop" not in expr.combined:
+                omp_expr += '\n{'
+        return omp_expr
 
     def _print_OMP_Single_Construct(self, expr):
         omp_expr   = str(expr.txt)
         return '#pragma omp {}\n{{'.format(omp_expr)
 
+    def _print_OMP_Critical_Construct(self, expr):
+        omp_expr = str(expr.txt)
+        omp_expr = '#pragma omp {}\n{{'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Master_Construct(self, expr):
+        omp_expr = str(expr.txt)
+        omp_expr = '#pragma omp {}\n{{'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Masked_Construct(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr = '#pragma omp masked{}'.format(clauses)
+        if expr.combined is None:
+            omp_expr += '\n{'
+        return omp_expr
+
+    def _print_OMP_Task_Construct(self, expr):
+        omp_expr = str(expr.txt)
+        omp_expr = '#pragma omp {}\n{{'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Target_Construct(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr = '#pragma omp target{}'.format(clauses)
+        if expr.combined is None:
+            omp_expr += '\n{'
+        elif (expr.combined and "for" not in expr.combined):
+            omp_expr += '\n{'
+        return omp_expr
+
+    def _print_OMP_Cancel_Construct(self, expr):
+        omp_expr = str(expr.txt)
+        omp_expr = '#pragma omp {}\n'.format(omp_expr)
+        return omp_expr
+
+    def _print_OMP_Teams_Construct(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr = '#pragma omp teams{}'.format(clauses)
+        if expr.combined is None:
+            omp_expr += '\n{'
+        return omp_expr
```

Comment:
Similar to above, could we have an `OmpCombinedAnnotatedComment` class to group these cases?